src/celeritas/phys/PhysicsStepUtils.hh

```diff
@@ -209,6 +209,12 @@ CELER_FUNCTION ParticleTrackView::Energy
         return Energy{0};
     }
 
+    if (particle.energy() < physics.scalars().eloss_calc_limit())
+    {
+        // Energy loss below the lowest e+/e- kinetic energy (1 keV)
+        return particle.energy();
+    }
```

Comment:
We originally had something like this, although we erroneously applied the cut as a post-processing step. Should this value reuse the CutoffParams so it can be independent on material and particle type? 

See  https://github.com/celeritas-project/celeritas/issues/295

Reply:
This one is just a constant cut with value `G4EmParameters::LowestElectronEnergy()`. Note that with this cut (and to a lesser extent because of energy loss straggling) we're going to again have the [problem](https://github.com/celeritas-project/celeritas/issues/354) of tracks coming to rest exactly on a boundary.

Reply:
@whokion Do you think it would be safe(r?) to wrap this in a `if (action != geo_boundary())`? I think that would solve the problem and potentially also reduce the chance of odd edge cases like in our grid-based energy deposition tally.

Reply:
(Excellent point @amandalund !)

Reply:
> @whokion Do you think it would be safe(r?) to wrap this in a `if (action != geo_boundary())`? I think that would solve the problem and potentially also reduce the chance of odd edge cases like in our grid-based energy deposition tally.

Where do you want to impose this additional condition? Inside calc_energy_loss? - i.e, adding an extra check, if (KE< eloss_calc_limit() && action != geo_boundary())? Doesn't geometry push particles on a volume boundary an extra epsilon at the end of geometry limited step? - i.e, should at-rest process be never happened exactly on a boundary at the beginning of stepping?

Reply:
Yeah, for this case it would be inside `calc_energy_loss` (although the `field-refactor` branch is changing all this anyway).

Yes, there is a hack in place to bump particles "backward" if they die on a boundary, but that won't work correctly if we have MSC or magnetic field enabled, because the end-of-step direction is not necessarily toward the beginning-of-step position. I think it's much better to avoid the unphysical bias of dying on a geometry boundary (e.g. emitting annihilation gammas on a surface).

Reply:
Actually since we don't have access to the track in the PhysicsStepUtils maybe move this piece of logic to the `AlongStep` code for now? (It doesn't matter where it goes because by monday it will be somewhere else :) )